Yank/experiment.py

```diff
@@ -2278,7 +2307,11 @@ def _run_experiment(self, experiment):
             switch_experiment_interval = self._options['switch_experiment_interval']
 
         built_experiment = self._build_experiment(experiment_path, experiment)
-        built_experiment.run(n_iterations=switch_experiment_interval)
+        # Trap a NaN'd simulation by capturing only the error we can handle, let all others raise normally
+        try:
+            built_experiment.run(n_iterations=switch_experiment_interval)
+        except utils.SimulationNaNError as e:
+            return e
```

Comment:
This may be redundant. The function should propagate the exception even without an explicit `try...except` clause.

Reply:
No, because here I `return e` not `raise e`, the former allows `run_experiments` to accept the error as a manipulatable object where as the later would just raise the error where it would have to be trapped up above too. I had thought about the `try..except` clause in `run_experiments`, but there was no way for me to trap the MPI distributed instance of it since each one could raise its own error. 

This block also ensures that any non `SimulationNaNError`s are raised as normal

Reply:
Ah, I understand now, thanks! That gives me an idea though, what do you think of making `mpi.distribute(catch_exceptions=True)` catching and returning the exception instead of the task result if something goes wrong? I don't think it'll be too hard to add.

Reply:
I'm not sure I follow, if any of the MPI jobs fail, wont the error propagate up the stack instead of the task returning a proper task result? What would a caught exception do?